grpc/naming.go

```diff
@@ -19,25 +19,39 @@ import (
 var (
 	// MinimumRefreshInterval decides the maximum sleep time between SRV Lookups, otherwise controlled by TTL of records.
 	MinimumRefreshInterval = 5 * time.Second
-	// MaximumConsecutiveErrors identifies how many iterations of bad SRV Lookups to tolerate in a loop.
-	MaximumConsecutiveErrors = 5
 )
 
 var errClosed = errors.New("go-srvlb: closed")
 
 // resolver implements the naming.Resolver interface from gRPC.
 type resolver struct {
 	srvResolver srv.Resolver
+	maxConsecutiveErrors int
+}
+
+type SrvResolverOptions func(*resolver)
+
+// MaximumConsecutiveErrors identifies how many consecutive iterations of bad SRV Lookups to tolerate in a loop.
+// After this limit is reached all the srv resolutions will stop.
+// Default value is -1. -1 means there is no limit.
+func WithMaximumConsecutiveErrors(maxErr int) SrvResolverOptions {
+	return func (resolver *resolver) {
+		resolver.maxConsecutiveErrors = maxErr
+	}
 }
 
 // New creates a gRPC naming.Resolver that is backed by an SRV lookup resolver.
-func New(srvResolver srv.Resolver) naming.Resolver {
-	return &resolver{srvResolver: srvResolver}
+func New(srvResolver srv.Resolver, options ...SrvResolverOptions) naming.Resolver {
```

Comment:
given that there's only one option, perhaps make it a required parameter which removes the need for the Option funcs